### PR TITLE
Verifying sensitivity

### PR DESCRIPTION
**Pull request by LucasFuentesValenzuela**
Goal of PR is to add some code to check that the sensitivity analysis actually works

### CODE REVIEW

**Comment by LucasFuentesValenzuela** [comment]
@degleris1 can you read a bit the code I added, and let me know what you think? 

disclaimer: my julia formatter does not work for some reason -- so will make it nicer to read later. 

In a nutshell, the results seem promising. I just saw one failure mode, which is for the storage variables wrt demand. I have to figure out why we don't seem to get a good value of sensitivity there.

File: src/CarbonNetworks.jl [hunk 1]
```diff
@@ -9,6 +9,7 @@ using LightGraphs
 using LinearAlgebra
 using SimpleWeightedGraphs
 using Zygote
+using Plots
```
Comment (degleris1) [inline src/CarbonNetworks.jl:12]: Let's not add `Plots` directly to the package. `Plots` takes a long time to compile. Instead, it is recommend we use `RecipesBase` (https://github.com/JuliaPlots/RecipesBase.jl).

File: src/dyn_model.jl [hunk 2]
```diff
@@ -141,6 +141,11 @@ Details:
 """
 kkt_dims_dyn(n, m, l, T) = T * (6n + 3m + 3l)
 
+"""
+add doc
+"""
+storage_kkt_dims(n) = 5n
```
Comment (degleris1) [inline src/dyn_model.jl:147]: Could we have `kkt_dims_dyn` use this function?

File: src/dyn_sensitivity.jl [hunk 3]
```diff
@@ -41,12 +41,176 @@ function compute_mefs(P::PowerManagementProblem, net::DynamicPowerNetwork, d, c,
...
     # Construct ∇_x C(x)
     ∇C_dyn = zeros(kkt_dims_dyn(n, m, l, T))
-    idx = 0
-    for _ in 1:T
-        ∇C_dyn[idx+1 : idx+l] .= c
-        idx += static_dim
-    end
-    
+    idx = (t-1)*static_dim
+    ∇C_dyn[idx+1 : idx+l] .= c
```
Comment (degleris1) [inline src/dyn_sensitivity.jl:45]: Could this be the bug? I might play with this a bit. Basically, the potential issue here is that the gradient of carbon cost is really dependent on generation at every timestep. If we provide a gradient that only reflects the dependency on a single timestep, we may produce an incorrect gradient with respect to demand.

File: src/model.jl [hunk 4]
```diff
@@ -28,6 +28,8 @@ mutable struct PowerManagementProblem
     params
 end
 
+OPT = () -> ECOS.Optimizer(verbose=false)
```
Comment (degleris1) [inline src/model.jl:31]: We can put this in `CarbonNetworks.jl`

File: src/dyn_sensitivity.jl (hunk 3 above)
Comment (degleris1) [reply src/dyn_sensitivity.jl:45]: Need to change this.

**Review by degleris1 — CHANGES_REQUESTED** [review]

File: src/dyn_sensitivity.jl (hunk 3 above)
Comment (LucasFuentesValenzuela) [reply src/dyn_sensitivity.jl:45]: @degleris1 do we agree that `c` is constant over time in our assumptions? And therefore does not need to be specified as an array of size `T`?

File: src/CarbonNetworks.jl (hunk 1 above)
Comment (LucasFuentesValenzuela) [reply src/CarbonNetworks.jl:12]: @degleris1 I am not sure I totally understand how this works. It looks like a macro/python decorator. Can you clarify what changes are required? e.g. in `plot_sensitivity_check` in `dyn_sensitivity.py`?

File: src/dyn_sensitivity.jl (hunk 3 above)
Comment (degleris1) [reply src/dyn_sensitivity.jl]: Agreed! I mean we could model it as varying, but I think it's unnecessary.

**Pull request merged**